pkg/providers/vmware/provider_test.go

```diff
@@ -0,0 +1,139 @@
+package vmware
+
+import (
+	"context"
+	"encoding/json"
+
+	vclient "github.com/kubevirt/vm-import-operator/pkg/providers/vmware/client"
+	vtemplates "github.com/kubevirt/vm-import-operator/pkg/providers/vmware/templates"
+	"github.com/kubevirt/vm-import-operator/pkg/templates"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	templatev1 "github.com/openshift/api/template/v1"
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/simulator"
+	"github.com/vmware/govmomi/vim25/mo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+)
+
+var _ = Describe("Processing a template", func() {
```

Comment:
Please implement tests for other `Provider` interface methods which are not no-ops.